crates/optimism/primitives/src/receipt.rs

```diff
@@ -120,6 +127,95 @@ impl OpReceipt {
             }
         }
     }
+
+    /// RLP-encodes receipt fields without an RLP header.
+    pub fn rlp_encode_fields_without_bloom(&self, out: &mut dyn BufMut) {
+        match self {
+            Self::Legacy(receipt) |
+            Self::Eip2930(receipt) |
+            Self::Eip1559(receipt) |
+            Self::Eip7702(receipt) => {
+                receipt.status.encode(out);
+                receipt.cumulative_gas_used.encode(out);
+                receipt.logs.encode(out);
+            }
+            Self::Deposit(receipt) => {
+                receipt.inner.status.encode(out);
+                receipt.inner.cumulative_gas_used.encode(out);
+                receipt.inner.logs.encode(out);
+                if let Some(nonce) = receipt.deposit_nonce {
+                    nonce.encode(out);
+                }
+                if let Some(version) = receipt.deposit_receipt_version {
+                    version.encode(out);
+                }
+            }
+        }
+    }
+
+    /// Returns length of RLP-encoded receipt fields without an RLP header.
+    pub fn rlp_encoded_fields_length_without_bloom(&self) -> usize {
+        match self {
+            Self::Legacy(receipt) |
+            Self::Eip2930(receipt) |
+            Self::Eip1559(receipt) |
+            Self::Eip7702(receipt) => {
+                receipt.status.length() +
+                    receipt.cumulative_gas_used.length() +
+                    receipt.logs.length()
+            }
+            Self::Deposit(receipt) => {
+                receipt.inner.status.length() +
+                    receipt.inner.cumulative_gas_used.length() +
+                    receipt.inner.logs.length() +
+                    receipt.deposit_nonce.map_or(0, |nonce| nonce.length()) +
+                    receipt.deposit_receipt_version.map_or(0, |version| version.length())
+            }
+        }
+    }
+
+    /// RLP-decodes the receipt from the provided buffer without bloom.
+    pub fn rlp_decode_inner_without_bloom(
+        buf: &mut &[u8],
+        tx_type: OpTxType,
+    ) -> alloy_rlp::Result<Self> {
+        let header = Header::decode(buf)?;
+        if !header.list {
+            return Err(alloy_rlp::Error::UnexpectedString);
+        }
+
+        let remaining = buf.len();
+        let status = Decodable::decode(buf)?;
+        let cumulative_gas_used = Decodable::decode(buf)?;
+        let logs = Decodable::decode(buf)?;
+
+        let mut deposit_nonce = None;
+        let mut deposit_receipt_version = None;
+
+        // For deposit receipts, try to decode nonce and version if they exist
+        if tx_type == OpTxType::Deposit && buf.len() + header.payload_length > remaining {
+            deposit_nonce = Some(Decodable::decode(buf)?);
+            if buf.len() + header.payload_length > remaining {
+                deposit_receipt_version = Some(Decodable::decode(buf)?);
+            }
+        }
```

Comment:
I'm actually not sure if we have the case where receipt version can be some but deposit nonce is None

which would be problematic, but this is already the case in the existing impl